### DISCUSSION TITLE
Bonsai error "Unable to acquire camera frame"

### DISCUSSION THREAD

**Discussion by NofarBart** [bonsai-rx/bonsai#1747] [Q&A] [answered]
Hi :)
I am an assistant in a neuroscience lab and would like to use your program to record behaving mice. I installed bonsai with the .exe, tried to stream (and record, also tried adding "VideoWriter" component and everytime got these messages: 

![WhatsApp Image 2024-04-17 at 11 00 45](https://github.com/bonsai-rx/bonsai/assets/83591641/0a2a6367-0c1f-4a34-8a97-93a1f821c99f)
![WhatsApp Image 2024-04-17 at 10 50 35](https://github.com/bonsai-rx/bonsai/assets/83591641/42ffb5bc-fad5-4a8e-a041-448339fccab5)

The recording camera is point grey Grasshopper3 GS3-U3-
23S6M.
Would appreciate any help identifing what is the problem,
Nofar

**Comment by bruno-f-cruz** [comment]
Hi,

You should use the [spinnaker package](https://www.nuget.org/packages/Bonsai.Spinnaker) if you want to interface with spinnaker cameras.
Hope it helps
Bruno

**Reply by NofarBart** [reply]
First of all thanks for the help!
I tried to install the spinnaker package and got this error-
![IMG_3778](https://github.com/bonsai-rx/bonsai/assets/83591641/f9fafe9b-38ff-4fa5-8ea4-244707681df7)
What do I need to do to make this work?
I also want to change the resolution to be 640x480, would appreciate any guidance 🙏🙏
Nofar

**Reply by NofarBart** [reply]
I see that the image isn't clear enough-
I used "Spinner Capture" as source and "VideoWriter" as sink. The error is "No method overload found for the given arguments"

**Comment by bruno-f-cruz** [comment]
You need to expose the image property from the camera operator. Right click on the operator and click image. Videowriter only works with images.

**Reply by NofarBart** [reply]
Hi Bruno, thank you for helping me out!

I tried to change the spinnaker's property and it is unaccessible (written in grey). Also tried some other things, including changing the videowriter into other operations and nothing seems to work. Now the software sends an error message of "A procedure imported be 'SpinnakerNET_v140.dll' could not be loaded". Can you help me find the source of the error?
Attaching photos to illustrate.
![WhatsApp Image 2024-05-06 at 14 09 24](https://github.com/bonsai-rx/bonsai/assets/83591641/91d3121e-a6ab-49c2-be5a-d6c4ae5bd9d6)
![WhatsApp Image 2024-05-06 at 14 09 30](https://github.com/bonsai-rx/bonsai/assets/83591641/50540883-9f86-47b4-8059-5002943d62e1)

Nofar

**Reply by bruno-f-cruz** [reply]
Did you install the SpinViewer version instructed in the package documentation?

**Reply by NofarBart** [reply]
Can you attach the documentation and a link to the SpinViewer version?
I didn't find it ): 
And also- do I need to write a code supporting the structure of operations or is setting the operations like this enough?
![example](https://github.com/bonsai-rx/bonsai/assets/83591641/344f3da5-e21e-42b2-a713-5f5884b45fc4)

**Reply by NofarBart** [reply]
It is unclear to me whether in order to record the camera output I need to write an algorithm, or it is suffice to use the icons "Spinnaker Capture" and "VideoWriter" for controlling the recording parameters.
![WhatsApp Image 2024-05-07 at 14 58 33](https://github.com/bonsai-rx/bonsai/assets/83591641/681f2721-fd60-458d-af50-6381d2eb7750)

**Comment by bruno-f-cruz** [comment, marked as answer]
As I stated before, `VideoWriter` only knows how to save images, not complex data types like a Spinnaker camera dataframe. Therefore, you must unpack the datatype returned by SpinnakerCapture and access the image portion. The general introduction to bonsai explains how to achieve this (https://bonsai-rx.org/docs/articles/editor.html#workflow).

Regarding the Spinnaker package dependencies, when you installed the package via the package manager, the description of the package states that a specific version of the SDK is required:
![image](https://github.com/bonsai-rx/bonsai/assets/7049351/7dc61d86-f91b-4e1b-864e-402bc00d2632)

**Reply by NofarBart** [reply]
Thank you, the SDK helped!
When I try to access the image:
![IMG_3871](https://github.com/bonsai-rx/bonsai/assets/83591641/50c80049-7652-4000-89e4-f6bd4999aab3)
I get this error:
![IMG_3922](https://github.com/bonsai-rx/bonsai/assets/83591641/dc709dad-7bed-411c-9736-84de83c5103c) (and a black video)
Also tried to see if the camera is connected and recording and got this:
![IMG_3923](https://github.com/bonsai-rx/bonsai/assets/83591641/9254f4fd-360b-48f9-a49f-d6c25ffec860)
Is there anything else I need to do to connect the camera to the workflow?
Nofar
